### PR TITLE
feat: deps usage should search in subdependencies

### PR DESCRIPTION
**Pull request by zkochan**
Related PRs:
- https://github.com/pnpm/pnpm/pull/7132

## Proposed Changes

- When pnpm is the package manager and dependencies are searched by package name, `bit deps usage pkg` should work as `pnpm why pkg`, searching for `pkg` in subdependencies, not only direct dependencies.
- Improve outputs to show the full hierarchy - 
  - [current output](https://app.warp.dev/block/w4ajnvQZRwAJMNZj93rNfh) 
  - [new output](https://app.warp.dev/block/xfERfDMsxwPvJi5cLIDnBZ)
- support `bit why <package name>` as alias for `bit deps usage <package name>`

### CODE REVIEW

File: scopes/docs/docs/docs.ui.runtime.tsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ export class DocsUI {
   private _emptyState?: ComponentType;
 
   registerEmptyState(emptyState: ComponentType) {
-    return this._emptyState = emptyState;
+    return (this._emptyState = emptyState);
```
Comment (GiladShoham) [inline scopes/docs/docs/docs.ui.runtime.tsx:35]: Please remove from this pr

File: scopes/docs/docs/overview/overview.tsx [hunk 2]
```diff
@@ -36,7 +36,7 @@ export type OverviewProps = {
   titleBadges: TitleBadgeSlot;
   overviewOptions: OverviewOptionsSlot;
   previewProps?: Partial<ComponentPreviewProps>;
-  getEmptyState?: () => ComponentType|undefined;
+  getEmptyState?: () => ComponentType | undefined;
```
Comment (GiladShoham) [inline scopes/docs/docs/overview/overview.tsx:39]: Please remove from this pr

File: scopes/docs/docs/overview/overview.tsx [hunk 3]
```diff
@@ -96,26 +96,28 @@ export function Overview({ titleBadges, overviewOptions, previewProps, getEmptyS
...
+            {...rest}
+            component={component}
+            style={{ width: '100%', height: '100%', minHeight: !isScaling ? 500 : undefined }}
+          />
+          {component.preview?.skipIncludes && <CompositionGallery isLoading={isLoading} component={component} />}
+          {component.preview?.skipIncludes && (
+            <PropertiesTable className={styles.overviewPropsTable} componentId={component.id.toString()} />
+          )}
+        </div>
+      )}
```
Comment (GiladShoham) [inline scopes/docs/docs/overview/overview.tsx:119]: Please remove from this pr

**Review by GiladShoham — CHANGES_REQUESTED** [review]

File: scopes/docs/docs/docs.ui.runtime.tsx (hunk 1 above)
Comment (zkochan) [reply scopes/docs/docs/docs.ui.runtime.tsx:35]: I did not do such changes. What is it? Autoformat?

File: scopes/dependencies/dependencies/dependencies-cmd.ts [hunk 4]
```diff
@@ -293,9 +295,16 @@ export class DependenciesUsageCmd implements Command {
...
 
-  constructor(private deps: DependenciesMain) {}
+  constructor(
+    private deps: DependenciesMain,
+    private depsResolver: DependencyResolverMain,
+    private workspace: Workspace
+  ) {}
 
   async report([depName]: [string]) {
+    if (this.depsResolver.getPackageManager()?.findUsages && !isComponentId(depName)) {
```
Comment (GiladShoham) [inline scopes/dependencies/dependencies/dependencies-cmd.ts:305]: This code should be moved to the main runtime

File: scopes/dependencies/dependencies/dependencies-cmd.ts [hunk 5]
```diff
@@ -306,6 +315,14 @@ export class DependenciesUsageCmd implements Command {
   }
 }
 
+function isComponentId(depName: string) {
```
Comment (GiladShoham) [inline scopes/dependencies/dependencies/dependencies-cmd.ts:318]: This code should be moved to the main runtime

**Review by GiladShoham — CHANGES_REQUESTED** [review]

File: scopes/dependencies/dependencies/dependencies-cmd.ts (hunk 5 above)
Comment (zkochan) [reply scopes/dependencies/dependencies/dependencies-cmd.ts:318]: isn't this just a utility function that doesn't need to be aware of any context? What benefit does it have from being part of the runtime class?

Comment (zkochan) [reply scopes/dependencies/dependencies/dependencies-cmd.ts]: I moved the function to main runtime file.

**Review by GiladShoham — APPROVED** [review]

**Pull request merged**